### PR TITLE
Added extension to check same day

### PR DESCRIPTION
**Pull request by leschlogl**
**Description**:
I've refactored some parts that use verification of the same day. I've put into an extension to reuse code in the entire application with simple method calling, avoiding "copying and pasting" of the same day logic. Also, I've created some unit tests for this.

**Steps to test this PR**:
1 - Use the app in the parts refactored or use the Unit Tests

### CODE REVIEW

**Comment by leschlogl** [comment]
@bwaresiak This is the new PR based on #503 

Thanks in advance!

File: Core/DateExtension.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
+//
+//  Debounce.swift
```
Comment (bwaresiak) [inline Core/DateExtension.swift]: Ah I've just noticed that - can you please make this match the filename. :) Thanks!

Comment (leschlogl) [reply Core/DateExtension.swift]: Sorry, my mistake! Solved!

**Comment by bwaresiak** [comment]
@leschlogl thank you! :)

This is tracked internally at: https://app.asana.com/0/856498667320406/1157331335297497

**Pull request merged**